metals/src/main/scala/scala/meta/internal/metals/JavaInteractiveSemanticdb.scala

```diff
@@ -227,38 +231,56 @@ case class JdkVersion(
 
 object JdkVersion {
 
-  def getJavaVersionFromJavaHome(
-      javaHome: AbsolutePath
+  def maybeJdkVersionFromJavaHome(
+      maybeJavaHome: Option[AbsolutePath]
   ): Option[JdkVersion] = {
-
-    def fromReleaseFile: Option[JdkVersion] = {
-      val releaseFile = javaHome.resolve("release")
-      if (releaseFile.exists) {
-        val properties = ConfigFactory.parseFile(
-          releaseFile.toFile,
-          ConfigParseOptions.defaults().setSyntax(ConfigSyntax.PROPERTIES)
-        )
-        try {
-          val version = properties
-            .getString("JAVA_VERSION")
-            .stripPrefix("\"")
-            .stripSuffix("\"")
-          JdkVersion.parse(version)
-        } catch {
-          case NonFatal(e) =>
-            scribe.error("Failed to read jdk version from `release` file", e)
-            None
-        }
-      } else None
+    maybeJavaHome.flatMap { javaHome =>
+      fromReleaseFile(javaHome).orElse {
+        ShellRunner
+          .runSync(
+            List("java", "-version"),
+            javaHome,
+            redirectErrorOutput = true,
+            maybeJavaHome = maybeJavaHome.map(_.toString())
+          )
+          .flatMap { javaVersionResponse =>
+            "\\d+\\.\\d+\\.\\d+".r
+              .findFirstIn(javaVersionResponse)
+              .flatMap(JdkVersion.parse)
```

Comment:
```suggestion
        val sbOut = new StringBuilder()
        val exit = ShellRunner
          .runSync(
            List("java", "-version"),
            javaHome,
            redirectErrorOutput = true,
            processOut = s => {
              sbOut.append(s)
              sbOut.append(Properties.lineSeparator)
           },
            maybeJavaHome = maybeJavaHome.map(_.toString())
          )
           if (exit == 0){
             val javaVersionResponse = sbOut.toString
             "\\d+\\.\\d+\\.\\d+".r
              .findFirstIn(javaVersionResponse)
              .flatMap(JdkVersion.parse)
           } else {
             None
           }
```

Also, if you move it to a separate function we could write proper unit tests for it, which would be safer.

Reply:
`runSync` is returning an `Option[String]`
So it is reading the `inputStream` of the `SystemProcess`
So if there is anything to be read from it, it returns that, otherwise, it returns `None`.
Then if the version regex cannot be detected in the String read from that, the output would again be None in the `fromShellRunner`. So this is delivering the same logic. Just added `JdkVersionSuite` to test and make sure.

Reply:
If using             

```scala
processOut = s => {
              sbOut.append(s)
              sbOut.append(Properties.lineSeparator)
           }, 

```
is the correct way of getting the process output, I can change the runSync to reflect that. But then, it would be side-effecting, wouldn’t it?
perhapse, I can move the `StringBuilder` in the runSync itself, then return a tuple of (exitCode, maybeOutputString) from it.

Reply:
It’s just that the test is pretty flaky. Sometimes it passes and sometimes it succeeds. Is it due to the incorrect way of getting the process output? If so, I can use the previous logic in runSync which was returning a future. Then the name also perhaps can change.

Reply:
> But then, it would be side-effecting, wouldn’t it?

We are running a process, so this is side effecting anyway and also in tooling I wouldn't worry too much about it. We should keep the function consistent with the other ones.